src/main/java/com/palantir/stash/stashbot/servlet/RepoConfigurationStatusServlet.java

```diff
@@ -0,0 +1,119 @@
+import com.atlassian.stash.util.Page;
+import com.atlassian.stash.util.PageRequest;
+import com.atlassian.stash.util.PageRequestImpl;
+import com.atlassian.webresource.api.assembler.PageBuilderService;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.stash.stashbot.config.ConfigurationPersistenceService;
+import com.palantir.stash.stashbot.jobtemplate.JobType;
+import com.palantir.stash.stashbot.logger.PluginLoggerFactory;
+import com.palantir.stash.stashbot.managers.JenkinsManager;
+import com.palantir.stash.stashbot.managers.PluginUserManager;
+import com.palantir.stash.stashbot.persistence.JenkinsServerConfiguration;
+import com.palantir.stash.stashbot.persistence.RepositoryConfiguration;
+
+public class RepoConfigurationStatusServlet extends HttpServlet {
+
+    /**
+     *
+     */
+    private static final long serialVersionUID = 1L;
+
+    private final RepositoryService repositoryService;
+    private final PullRequestService prs;
+    private final SoyTemplateRenderer soyTemplateRenderer;
+    private final PageBuilderService pageBuilderService;
+    private final ConfigurationPersistenceService configurationPersistanceManager;
+    private final JenkinsManager jenkinsManager;
+    private final PluginUserManager pluginUserManager;
+    private final PermissionValidationService permissionValidationService;
+    private final Logger log;
+
+    public RepoConfigurationStatusServlet(RepositoryService repositoryService, PullRequestService prs,
+        SoyTemplateRenderer soyTemplateRenderer, PageBuilderService pageBuilderService,
+        ConfigurationPersistenceService configurationPersistenceManager, JenkinsManager jenkinsManager,
+        PluginUserManager pluginUserManager, PermissionValidationService permissionValidationService,
+        PluginLoggerFactory lf) {
+        this.repositoryService = repositoryService;
+        this.prs = prs;
+        this.soyTemplateRenderer = soyTemplateRenderer;
+        this.pageBuilderService = pageBuilderService;
+        this.configurationPersistanceManager = configurationPersistenceManager;
+        this.jenkinsManager = jenkinsManager;
+        this.permissionValidationService = permissionValidationService;
+        this.pluginUserManager = pluginUserManager;
+        this.log = lf.getLoggerForThis(this);
+    }
+
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+        Repository rep = getRepository(req);
+        if (rep == null) {
+            res.sendError(404);
+            return;
+        }
+        RepositoryConfiguration rc;
+        try {
+            rc = configurationPersistanceManager.getRepositoryConfigurationForRepository(rep);
+        } catch (SQLException e1) {
+            throw new ServletException(e1);
+        }
+        res.setContentType("text/html;charset=UTF-8");
+        res.getWriter().print(rc.getCiEnabled());
```

Comment:
possibly cleaner to have this return a JSON object and parse it as such on the front-end?